stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb

```diff
@@ -807,3 +807,17 @@ extension CGFloat : _CVarArgPassedAsDouble, _CVarArgAligned {
     return native._cVarArgAlignment
   }
 }
+
+extension CGFloat : Codable {
+  @_transparent
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.singleValueContainer()
+    self.native = try container.decode(NativeType.self)
```

Comment:
Potentially, yes, though we face a similar portability issue with `Int`... One thing we can do is always encode as a `Double` and simply fail on 32-bit platforms if the decoded `Double` does not fit in a `Float`. This doesn't change anything about the failure, but may play more nicely with fixed-width format `Encoder`s and `Decoder`s which may prevent you from decoding a `Double` if you encoded a `Float`, or vice versa (i.e. we push the failure out from `Decoder` code into `CGFloat` code, which we control, and can provide an error message). @parkera Thoughts?